.github/workflows/test.yml

```diff
@@ -52,11 +52,11 @@ jobs:
         run: |
           # NOTE: Install with [all] so we can import plugins which may
           #       have extra dependencies.
-          pip install  \
+          pip install \
             "cylc-flow[all] @ git+https://github.com/cylc/cylc-flow@${cylc_flow}" \
-            "cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@${cylc_rose}" \
             "cylc-uiserver[all] @ git+https://github.com/cylc/cylc-uiserver@${cylc_uis}" \
-            "metomi-rose[all] @ git+https://github.com/metomi/rose@${meto_rose}"
+            "metomi-rose[all] @ git+https://github.com/metomi/rose@${meto_rose}" \
+            "cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@${cylc_rose}"
```

Comment:
Gah, what a waste of 1.5 hours 😤

Reply:
And yet the original order works fine here 🤔🤔🤔: https://github.com/cylc/cylc-flow/blob/dfe4208574c5c750f7c49780d011e8d849ed81d3/.github/workflows/test_tutorial_workflow.yml#L45-L47

Reply:
Did these failures happen before the cylc-flow/metomi-rose dependencies were bumped in both the cylc-rose and cylc-uiserver repos?

If not pip might have tried to pull in compatible versions from pypi.

Reply:
Actually it looks like cylc-rose was installed *before* metomi-rose so this was bad before.

Reply:
The error I was getting was suggesting it was trying to pull in `metomi-rose` from PyPI
```
ERROR: Could not find a version that satisfies the requirement metomi-rose==2.0rc2.*
        (from cylc-rose[all]) (from versions: 2.0a1, 2.0b1, 2.0b2, 2.0b3, 2.0rc1)
ERROR: No matching distribution found for metomi-rose==2.0rc2.*
```
Are you saying: before, when we did `pip install cylc-rose @ github` followed by `git clone metomi-rose`, it was installing `metomi-rose` from PyPI in the first step and replacing it in the second?

Reply:
Yep, horrible.